api/datadoghq/v1alpha2/datadogpodautoscaler_types.go

```diff
@@ -0,0 +1,542 @@
+//
+
+// DatadogPodAutoscalerScalingStrategySelect is used to specify which policy should be used while scaling in a certain direction
+// +kubebuilder:validation:Enum:=Max;Min;Disabled
+type DatadogPodAutoscalerScalingStrategySelect string
+
+const (
+	// DatadogPodAutoscalerMaxChangeStrategySelect selects the policy with the highest possible change.
+	DatadogPodAutoscalerMaxChangeStrategySelect DatadogPodAutoscalerScalingStrategySelect = "Max"
+
+	// DatadogPodAutoscalerMinChangeStrategySelect selects the policy with the lowest possible change.
+	DatadogPodAutoscalerMinChangeStrategySelect DatadogPodAutoscalerScalingStrategySelect = "Min"
+
+	// DatadogPodAutoscalerDisabledStrategySelect disables the scaling in this direction.
+	DatadogPodAutoscalerDisabledStrategySelect DatadogPodAutoscalerScalingStrategySelect = "Disabled"
+)
+
+// DatadogPodAutoscalerScalingPolicy defines the policy to scale the target resource.
+type DatadogPodAutoscalerScalingPolicy struct {
+	// Strategy is used to specify which policy should be used.
+	// If not set, the default value Max is used.
+	// +optional
+	Strategy *DatadogPodAutoscalerScalingStrategySelect `json:"strategy,omitempty"`
+
+	// Rules is a list of potential scaling polices which can be used during scaling.
+	// At least one policy must be specified, otherwise the DatadogPodAutoscalerScalingPolicy will be discarded as invalid
+	// +listType=atomic
+	// +optional
+	Rules []DatadogPodAutoscalerScalingRule `json:"rules,omitempty"`
+
+	// StabilizationWindowSeconds is the number of seconds the controller should lookback at previous recommendations
+	// before deciding to apply a new one. Defaults to 0.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=1800
+	StabilizationWindowSeconds int32 `json:"stabilizationWindowSeconds,omitempty"`
+}
+
+// DatadogPodAutoscalerScalingRuleType defines how scaling rule value should be interpreted.
+// +kubebuilder:validation:Enum:=Pods;Percent
+type DatadogPodAutoscalerScalingRuleType string
+
+const (
+	// DatadogPodAutoscalerPodsScalingRuleType specifies a change in the absolute number of pods compared to the starting number of pods.
+	DatadogPodAutoscalerPodsScalingRuleType DatadogPodAutoscalerScalingRuleType = "Pods"
+
+	// DatadogPodAutoscalerPercentScalingRuleType specifies a relative amount of change compared to the starting number of pods.
+	DatadogPodAutoscalerPercentScalingRuleType DatadogPodAutoscalerScalingRuleType = "Percent"
+)
+
+// DatadogPodAutoscalerScalingRule defines rules for horizontal scaling that should be true for a certain amount of time.
+type DatadogPodAutoscalerScalingRule struct {
+	// Type is used to specify the scaling policy.
+	Type DatadogPodAutoscalerScalingRuleType `json:"type"`
+
+	// Value contains the amount of change which is permitted by the policy.
+	// Setting it to 0 will prevent any scaling in this direction and should not be used unless Match is set to IfScalingEvent.
```

Comment:
should this line be removed? It doesn't look like the `Match` constant is present in this api

Reply:
Dropped the part about setting Match.